### ISSUE TITLE
[java] ProperLogger gives false positives for inner classes

### ISSUE THREAD

**Issue by cristalp** [pmd/pmd#3436]
**Affects PMD Version:** 6.29.0

**Rule:** ProperLogger
https://pmd.github.io/latest/pmd_rules_java_errorprone.html#properlogger

**Description:**
When checking inner classes, this rule gives false positives.

**Code Sample demonstrating the issue:**

```
public class ResultHolders {

  private ResultHolders() {
    super();
  }

  private static class SynchronizedHolder<T extends Ticket> implements ResultHolder<T> {

    private static final Logger LOG = LoggerFactory.getLogger(SynchronizedHolder.class);

  }

  private static class JMSHolder<T extends Ticket> implements ResultHolder<T> {

    private static final Logger LOG = LoggerFactory.getLogger(JMSHolder.class);

  }

  static class ResultMessage {

    private static final Logger LOG = LoggerFactory.getLogger(ResultMessage.class);

  }

}
```

**Expected outcome:** Should not report an error, but reports an error for every Logger.

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
